//Message format for transmitting commands
// from UI application to LC.
syntax = "proto3";

package surtrpb;

message SwitchControl {
    int32 id = 1;
    bool state = 2;
}

message StepperControl {
    int32 id = 1;
    sint32 motorDelta = 2;
}

message Ignition {
    int32 password = 1;
}

message ADCMeasurement {
    int32 id = 1;
    int32 value = 2;
}

message SurtrMessage {
    optional int64 us_since_boot = 1;
    oneof command {
        SwitchControl sw_ctrl = 2;
        StepperControl step_ctrl = 3;
        ADCMeasurement adc_measurement = 4;
        Ignition ignition = 5;
    };
}